### PR TITLE
[15.0][MIG] product_multi_price: Migration to version 15.0

### PR DESCRIPTION
**Pull request by ernesto-garcia-tecnativa**
@tecnativa TT37298

### CODE REVIEW

**Comment by ernesto-garcia-tecnativa** [comment]
@pedrobaeza @chienandalu please review

**Review by pedrobaeza — APPROVED** [review]

**Comment by ernesto-garcia-tecnativa** [comment]
@pedrobaeza please review

**Comment by pedrobaeza** [comment]
I already reviewed it. I have tagged @chienandalu, which is the original author.

**Review by chienandalu — COMMENTED** [review]
Testing in runbot I get this error when I try to configure a price in a template or variant:

```
Traceback (most recent call last):
  File "/opt/odoo/odoo/addons/base/models/ir_http.py", line 237, in _dispatch
    result = request.dispatch()
  File "/opt/odoo/odoo/http.py", line 687, in dispatch
    result = self._call_function(**self.params)
  File "/opt/odoo/odoo/http.py", line 359, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/opt/odoo/odoo/service/model.py", line 94, in wrapper
    return f(dbname, *args, **kwargs)
  File "/opt/odoo/odoo/http.py", line 348, in checked_call
    result = self.endpoint(*a, **kw)
  File "/opt/odoo/odoo/http.py", line 910, in __call__
    return self.method(*args, **kw)
  File "/opt/odoo/odoo/http.py", line 535, in response_wrap
    response = f(*args, **kw)
  File "/opt/odoo/addons/web/controllers/main.py", line 1326, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/opt/odoo/addons/web/controllers/main.py", line 1318, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/opt/odoo/odoo/api.py", line 464, in call_kw
    result = _call_kw_multi(method, model, args, kwargs)
  File "/opt/odoo/odoo/api.py", line 451, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/opt/odoo/odoo/models.py", line 3225, in read
    self._read(stored_fields)
  File "/opt/odoo/odoo/models.py", line 3343, in _read
    cr.execute(query_str, params + [sub_ids])
  File "<decorator-gen-3>", line 2, in execute
  File "/opt/odoo/odoo/sql_db.py", line 90, in check
    return f(self, *args, **kwargs)
  File "/opt/odoo/odoo/sql_db.py", line 311, in execute
    res = self._obj.execute(query, params)
Exception

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/odoo/odoo/http.py", line 643, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/opt/odoo/odoo/http.py", line 301, in _handle_exception
    raise exception.with_traceback(None) from new_cause
psycopg2.errors.DatatypeMismatch: COALESCE types text and integer cannot be matched
LINE 1: ...d", COALESCE("product_multi_price__name"."value", "product_m...
```

**Comment by ernesto-garcia-tecnativa** [comment]
> Testing in runbot I get this error when I try to configure a price in a template or variant:
> 
> ```
> Traceback (most recent call last):
>   File "/opt/odoo/odoo/addons/base/models/ir_http.py", line 237, in _dispatch
>     result = request.dispatch()
>   File "/opt/odoo/odoo/http.py", line 687, in dispatch
>     result = self._call_function(**self.params)
>   File "/opt/odoo/odoo/http.py", line 359, in _call_function
>     return checked_call(self.db, *args, **kwargs)
>   File "/opt/odoo/odoo/service/model.py", line 94, in wrapper
>     return f(dbname, *args, **kwargs)
>   File "/opt/odoo/odoo/http.py", line 348, in checked_call
>     result = self.endpoint(*a, **kw)
>   File "/opt/odoo/odoo/http.py", line 910, in __call__
>     return self.method(*args, **kw)
>   File "/opt/odoo/odoo/http.py", line 535, in response_wrap
>     response = f(*args, **kw)
>   File "/opt/odoo/addons/web/controllers/main.py", line 1326, in call_kw
>     return self._call_kw(model, method, args, kwargs)
>   File "/opt/odoo/addons/web/controllers/main.py", line 1318, in _call_kw
>     return call_kw(request.env[model], method, args, kwargs)
>   File "/opt/odoo/odoo/api.py", line 464, in call_kw
>     result = _call_kw_multi(method, model, args, kwargs)
>   File "/opt/odoo/odoo/api.py", line 451, in _call_kw_multi
>     result = method(recs, *args, **kwargs)
>   File "/opt/odoo/odoo/models.py", line 3225, in read
>     self._read(stored_fields)
>   File "/opt/odoo/odoo/models.py", line 3343, in _read
>     cr.execute(query_str, params + [sub_ids])
>   File "<decorator-gen-3>", line 2, in execute
>   File "/opt/odoo/odoo/sql_db.py", line 90, in check
>     return f(self, *args, **kwargs)
>   File "/opt/odoo/odoo/sql_db.py", line 311, in execute
>     res = self._obj.execute(query, params)
> Exception
> 
> The above exception was the direct cause of the following exception:
> 
> Traceback (most recent call last):
>   File "/opt/odoo/odoo/http.py", line 643, in _handle_exception
>     return super(JsonRequest, self)._handle_exception(exception)
>   File "/opt/odoo/odoo/http.py", line 301, in _handle_exception
>     raise exception.with_traceback(None) from new_cause
> psycopg2.errors.DatatypeMismatch: COALESCE types text and integer cannot be matched
> LINE 1: ...d", COALESCE("product_multi_price__name"."value", "product_m...
> ```

@chienandalu please review

**Comment by ernesto-garcia-tecnativa** [comment]
@chienandalu please review

File: product_multi_price/models/product_multi_price.py [hunk 1]
```diff
@@ -5,13 +5,11 @@
 
 class ProductMultiPrice(models.Model):
     _name = "product.multi.price"
+    _rec_name = "name_text"
     _description = "Product Multiple Prices"
 
-    name = fields.Many2one(
-        comodel_name="product.multi.price.name",
-        required=True,
-        translate=True,
```
Comment (pedrobaeza) [inline product_multi_price/models/product_multi_price.py]: Isn't maybe the problem just the `translate=True` thing and you don't need to do the rest of the changes?

Comment (ernesto-garcia-tecnativa) [reply product_multi_price/models/product_multi_price.py]: i tried what you said but i didn't get the expected result

Comment (pedrobaeza) [reply product_multi_price/models/product_multi_price.py]: So did you remove that attribute and the error continues?

Comment (ernesto-garcia-tecnativa) [reply product_multi_price/models/product_multi_price.py]: I tried both cases with and without attribute _translated_ and the error its present when the field 'name' is type Many2one

Comment (chienandalu) [reply product_multi_price/models/product_multi_price.py]: Ouh, that might be the key to it. `name` is a special property in `models.Model`. Try to make the many2one a different field and `name` to be a Char computed.

Comment (pedrobaeza) [reply product_multi_price/models/product_multi_price.py]: Please check, that it has been already corrected changing `_rec_name`.

Comment (chienandalu) [reply product_multi_price/models/product_multi_price.py]: Oh, right :) And now is working indeed :sweat_smile: Let me test it a little bit for a final review

**Review by chienandalu — APPROVED** [review]
Tested :+1:

**Pull request merged**